### PR TITLE
Generate V1-style date suggestions in the [[ menu & command palette

### PR DESCRIPTION
**Pull request by maccman**
## What

Typing `[[3 days ago`, `[[next friday`, `[[tomorrow`, `[[12/25`, or `[[December 2nd` now surfaces the matching **daily note** — synthesised from the query and lazily created on link — in both the editor's `[[` autocomplete and the command palette.

Before this, V2 only recognised a full ISO `YYYY-MM-DD`, so linking to a date meant typing the exact string. This ports the distinctive date-generator behaviour of the original Reflect; see [`docs/reflect-v1-backlink-menu.md`](docs/reflect-v1-backlink-menu.md) for the spec.

## Why

`[[Wiki Links]]` + daily-notes-first is core to Reflect, and "link to a date you've never opened" is a defining flow. ISO-only recognition made it the one wiki-link path that required knowing the exact canonical string — a clear parity gap with V1.

## How

**Core**
- New pure, dependency-free generator [`packages/core/src/indexing/date-suggestions.ts`](packages/core/src/indexing/date-suggestions.ts) interpreting the query four ways and merging by resolved day (cap 3):
  - **Relative offsets** — `3 days ago`, `one day`, bare `1`; spelled `one`–`ten`; ~15-year sanity cap (so `17 years` / `1000 years` stay quiet); guarded so a month-name day number never becomes "2 days from now".
  - **Natural language** — `today`/`yesterday`/`tomorrow`; `this|next|last` + weekday/`week`/`weekend`/`month`; prefix-matched (`mon`, `next fri`).
  - **Typed calendar** — ISO `2026-06-19` and slash `12/25` / `23/2/2023` per the date-format preference, with ambiguous `12/10` → both readings.
  - **Month-name** — `December 2nd`, `2 Dec`, abbreviations, malformed ordinals (`31th`), optional year.
  - The clock is injected as `today` and all arithmetic runs in **UTC**, so DST can't skip or repeat a day (no `date-fns` dependency added to core).
- `WikiSuggestion` gains an optional `phrase`; pure `mergeDateSuggestions` folds dates into the ranked index results: **dates lead, but an exact title/alias match keeps the top slot**, and an existing daily is reused so the link resolves to its real file (no duplicate day).
- `suggestWikiTargets(query, limit, dateGen?)` takes an opt-in context; the no-`dateGen` path is unchanged.

**Editor `[[` menu** — passes the clock + date-format, renders the phrase as the label and the resolved day as the detail, and suppresses the "Create …" row when the query reads as a date. Also fixed a stale comment: meowdown `0.8.1` hands the handler the **raw** post-`[[` text (case/spaces/slashes preserved), not a "lowercased, punctuation-stripped" query.

**Command palette** — opts in too; generated dates are appended **after** real note matches (in a global open surface your notes should outrank "Next Monday", unlike the `[[` menu where dates lead). The phrase renders above a muted resolved date; path-less dailies were already jumpable via the existing `dailyPath` synthesis.

## Reviewer notes — deliberate divergences from V1 (documented in tests)
- **Ambiguous shorthand** (`12/10`) uses the **current year for both** readings (V1's mixed Oct-2020/Dec-2019 looked like an idiosyncrasy).
- **Relative phrases always use digits** ("1 day ago"), so `[[one day ago]]` reads "1 day ago" (V1's spelled-echo came from a separate NL engine that isn't ported).
- **`this/next/last week`** uses a fixed **Monday start** in this PR. _Correction:_ V2 **does** have a `weekStartDay` setting — the follow-up [#229](https://github.com/team-reflect/reflect-open/pull/229) makes these phrases respect it.

## Testing
- `pnpm typecheck` clean; `pnpm lint` 0 errors (the 4 warnings are pre-existing, in unrelated files).
- New tests green: **21** generator cases mirror the V1 doc's worked-examples table (today = 2020-01-01, `dmy`), plus `mergeDateSuggestions` ordering/reuse/cap, editor create-suppression, and palette-append coverage.
- Verified via unit tests + typecheck/lint; **not** yet run in a live `tauri dev` session.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

---

> [!NOTE]
> **Low Risk**
> Additive autocomplete/indexing behavior behind an opt-in API; legacy `suggestWikiTargets` callers unchanged. Main risk is date-parsing edge cases or wrong day near midnight, mitigated by tests and `today` in the cache key.
> 
> **Overview**
> Ports **V1-style daily-note suggestions** from natural-language and partial date queries (e.g. `3 days ago`, `next friday`, `12/25`, `December 2nd`) into the **`[[` autocomplete** and **command palette**, not only full ISO `YYYY-MM-DD`.
> 
> **Core:** New pure `generateDateSuggestions` (injected `today` + `dateFormat`, UTC math, cap 3) and optional `phrase` on `WikiSuggestion`. `suggestWikiTargets` accepts an opt-in `DateSuggestionContext`; without it, behavior stays the same (ISO daily injection only). `mergeDateSuggestions` puts generated dates first in the `[[` menu but keeps an exact title/alias match on top and reuses existing daily rows.
> 
> **Desktop:** Editor and palette pass clock + format into `suggestWikiTargets`; palette query cache keys include `today` so relative labels don’t go stale after midnight. Palette **reorders** suggestions so real notes beat generated dates (opposite of `[[`). UI shows **phrase** as primary label with formatted date as detail; **Create** is suppressed when a generated date suggestion is present.
> 
> <sup>Reviewed by [Cursor Bugbot](https://cursor.com/bugbot) for commit fa0b852505989d0091ed1ec6895497d3930455c6. Bugbot is set up for automated code reviews on this repo. Configure [here](https://www.cursor.com/dashboard/bugbot).</sup>

### CODE REVIEW

File: apps/desktop/src/components/command-palette/use-palette-results.ts [hunk 1]
```diff
@@ -55,8 +56,9 @@ export function usePaletteResults(open: boolean, query: string): PaletteResults
     isLoading: suggestionsLoading,
     isError: suggestionsError,
   } = useQuery({
-    queryKey: [INDEX_QUERY_SCOPE, graph?.root, 'palette-suggest', trimmed],
-    queryFn: () => suggestWikiTargets(trimmed, 8),
+    queryKey: [INDEX_QUERY_SCOPE, graph?.root, 'palette-suggest', trimmed, settings.dateFormat],
+    queryFn: () =>
+      suggestWikiTargets(trimmed, 8, { today: todayIso(), dateFormat: settings.dateFormat }),
```
[the remark this answers is not in the document]
Comment (maccman) [reply apps/desktop/src/components/command-palette/use-palette-results.ts]: Fixed in 6e589ac — added `today` (todayIso()) to the palette-suggest query key, computed once so the key and queryFn agree on the same calendar day. Relative dates like "Tomorrow" no longer survive a midnight rollover in the cache.
